src/nvc.rs

```diff
@@ -2,6 +2,100 @@ use instruction::*;
 use interconnect::*;
 
 use std::collections::HashSet;
+use std::fmt;
+
+#[derive(Copy, Clone)]
+pub struct NvcCacheEntry {
+    pub tag: u32,
+    pub base_addr: u32,
+    pub subblock_valid: [bool; 2],
+}
+
+impl NvcCacheEntry {
+    fn new() -> NvcCacheEntry {
+        NvcCacheEntry {
+          tag: 0,
+          base_addr: 0,
+          subblock_valid: [false; 2],
+        }
+    }
+}
+
+impl fmt::Display for NvcCacheEntry {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "base_addr: 0x{:08x}, tag: 0x{:08x}, sb1 valid: {:5}, sb2 valid: {:5}", self.base_addr, self.tag, self.subblock_valid[0], self.subblock_valid[1])
+    }
+}
+
+pub struct NvcCache {
+    hits: usize,
+    misses: usize,
+    cache_entries: [NvcCacheEntry; 128],
+}
+
+impl NvcCache {
+    pub fn new() -> NvcCache {
```

Comment:
Since both this constructor and `NvcCacheEntry`'s constructor both seem to use more or less default values for all of their fields, having both derive the `Default` trait (done by annotating the struct with `#[derive(Default)]`) and initializing using the then-generated `NvcCache::default()` rather than `NvcCache::new()` should work here, and let us delete some code.

For an example of how to do this, check out e.g. https://github.com/emu-rs/rustual-boy/blob/master/src/vsu/mod.rs#L137 and https://github.com/emu-rs/rustual-boy/blob/master/src/vsu/mod.rs#L508 (and note that this only applies if all fields for a struct also have types that derive `Default`, but that should be the case here).